mitxgraders/comparers/comparers.py

```diff
@@ -178,3 +180,131 @@ def eigenvector_comparer(comparer_params_evals, student_eval, utils):
         }
 
     return utils.within_tolerance(actual, expected)
+
+def vector_span_comparer(comparer_params_evals, student_eval, utils):
+    """
+    Check whether student's answer is nonzero and in the span of some given
+    vectors.
+
+    comparer_params: A list of vectors
+
+    Usage
+    =====
+
+    Use a single vector as comparer_params to test whether student input is
+    parallel to a particular vector:
+    >>> from mitxgraders import MatrixGrader
+    >>> grader = MatrixGrader(
+    ...     answers={
+    ...         'comparer_params': [
+    ...             '[3, x, 1 + i]',
+    ...         ],
+    ...         'comparer': vector_span_comparer
+    ...     },
+    ...     variables=['x'],
+    ... )
+    >>> grader(None, '[3, x, 1 + i]')['ok']
+    True
+    >>> grader(None, '[9, 3*x, 3 + 3*i]')['ok']
+    True
+    >>> grader(None, '[9, 3*x, 3 - 3*i]')['ok']
+    False
+
+    Complex scale factors work, too:
+    >>> grader(None, '(4 + 2*i)*[3, x, 1 + i]')['ok']
+    True
+
+    Student input should be nonzero:
+    >>> result = grader(None, '[0, 0, 0]')
+    >>> result['ok']
+    False
+    >>> result['msg']
+    'Input should be a nonzero vector.'
+
+    Input shape is validated:
+    >>> grader(None, '5')
+    Traceback (most recent call last):
+    InputTypeError: Expected answer to be a vector, but input is a scalar
+
+    Multiple vectors can be provided:
+    >>> grader = MatrixGrader(
+    ...     answers={
+    ...         'comparer_params': [
+    ...             '[1, 1, 0]',    # v0
+    ...             '[0, 1, 2]'     # v1
+    ...         ],
+    ...         'comparer': vector_span_comparer
+    ...     },
+    ... )
+
+    The vector 2*v0 + 3i*v1 = [2, 2+3i, 6i] is in the span of v0 and v1:
+    >>> grader(None, '[2, 2 + 3*i, 6*i]')['ok']
+    True
+    """
+
+    # All comparer_param_evals should have same shape, or numpy.linalg.lstsq
+    # will let us know.
+    utils.validate_shape(student_eval, comparer_params_evals[0].shape)
+
+    if utils.within_tolerance(0, np.linalg.norm(student_eval)):
+        return {
+            'ok': False,
+            'grade_decimal': 0,
+            'msg': 'Input should be a nonzero vector.'
+        }
+
+    # Use ordinary least squares to find an approximation to student_eval
+    # that lies within the span of given vectors, then check that the
+    # residual-sum is small in comparison to student input.
+    column_vectors = np.array(comparer_params_evals).transpose()
+    ols = np.linalg.lstsq(column_vectors, student_eval)
```

Comment:
When we move to a newer version of numpy, we might want to add `rcond=None` to this.

Reply:
So I honestly can't remember why I wrote `rcond=-1`. That seems to be the default for our version of numpy anyway:

https://docs.scipy.org/doc/numpy-1.6.0/reference/generated/numpy.linalg.lstsq.html

The new default value (np 1.14+) should be only slightly bigge for our low-dimensional matrices:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.lstsq.html

@jolyonb Maybe we could just let numpy do it's defaults?

With unspecified `rcond`, my (fairly minimal) tests pass.

Reply:
I'm happy to leave this argument off for the moment, as things seem to work. We're expecting edX to upgrade to python 3.6 and the latest version of numpy+scipy later this year, so just flagging this as something that may need attention. Perhaps just leave a comment?